### DISCUSSION TITLE
Set custom analyzer on mapping

### DISCUSSION THREAD

**Discussion by bartpijn** [wallneradam/esorm#23] [Q&A] [answered]
Hi,

Great project, had this idea in mind for a long time myself but never came around to it.

I was wondering if there is a possibility to set custom tokenizers/analyzers on the ES mapping that is generated from the model?

Thanks in advance!

**Comment by wallneradam** [comment, marked as answer]
Hi, Thank you!

Because this library uses official ElasticSearch python library (uses AsyncElasticsearch from it), everything works possible with that also possible with this.

You can add additional settings to any models with `ESConfig` class in the `settings` property: https://github.com/wallneradam/esorm?tab=readme-ov-file#model-settings
You can set anything you found in ES documents: https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-custom-analyzer.html

Hope that helps
